tests/test_types.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from sys import version_info
+from typing import Optional, Union
+
+from pytest import mark, raises
+
+from pydargs import parse
+
+
+class TestUnion:
+    @dataclass
+    class Config:
+        a: int
+        b: Optional[int] = None
+        c: Union[int, str] = "b"
+        d: Union[None, int] = None
+        e: Optional[Union[int, float]] = None
+        f: float = 2.0
+
+    def test_default(self):
+        config = parse(self.Config, ["--a", "1"])
+        assert config.a == 1
+        assert config.b is None
+        assert config.c == "b"
+        assert config.d is None
+        assert config.e is None
+
+    def test_parse_optional(self):
+        config = parse(self.Config, ["--a", "1", "--b", "2"])
+        assert config.b == 2
```

Comment:
From this test case, I infer that we are using the first succeeding type cast as the final type. Maybe we should mention this in the readme, or is this obvious?

Reply:
I've expanded the README. Please have a look